fix: can't add gitlab personal access token and add more debug log in validate_provider_token

littlebai3618 opened the pull request:
- [ ] This change is worth documenting at https://docs.all-hands.dev/
- [ ] Include this change in the Release Notes. If checked, you **must** provide an **end-user friendly** description for your change below

**End-user friendly description of the problem this fixes or functionality this introduces.**
When the GitLab server’s /v4/user endpoint returns an empty avatar_url, token validation fails.


---
**Summarize what the PR does, explaining any non-trivial design decisions.**

In some self-hosted GitLab instances, the avatar_url field can be returned as None.
When this happens, GitLabService.get_user fails while building a User object, because the User model expects avatar_url to be a string, triggering a pydantic_core._pydantic_core.ValidationError (see stack trace below).

```shell
Traceback (most recent call last):
  File "/Users/baii/pycharmProjects/beike/OpenHands/openhands/integrations/utils.py", line 39, in validate_provider_token
    await gitlab_service.get_user()
  File "/Users/baii/pycharmProjects/beike/OpenHands/openhands/integrations/gitlab/gitlab_service.py", line 171, in get_user
    return User(
           ^^^^^
  File "/Users/baii/Library/Caches/pypoetry/virtualenvs/openhands-ai-394k-RCl-py3.12/lib/python3.12/site-packages/pydantic/main.py", line 253, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 1 validation error for User
avatar_url
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.11/v/string_type
```

I fixed the issue by adding a check for avatar_url in the method and supplying a default value of empty string

This minimal change resolves the problem in my tests.
I also added two debug log lines to validate_provider_token to make troubleshooting easier.


During the debugging process, I also noticed that the /user/info endpoint and the runtime image build were pulling the user repository using the default gitlab.com instead of the specified GitLab domain. I have fixed both issues.

Please carefully review whether the ProviderToken.host property has a default value when the user does not provide a domain. I'm not very familiar with this repository's code。

---
**Link of any specific issues this addresses:**
https://github.com/All-Hands-AI/OpenHands/issues/8781
https://github.com/All-Hands-AI/OpenHands/issues/8534

malhotra5 commented on openhands/runtime/base.py:
```diff
@@ -360,12 +360,7 @@ async def clone_or_init_repo(
-        }
-
-        domain = provider_domains[provider]
+        domain = git_provider_tokens[provider].host
```
```suggestion
        provider_domains = {
            ProviderType.GITHUB: 'github.com',
            ProviderType.GITLAB: 'gitlab.com',
        }
        domain = git_provider_tokens[provider].host if git_provider_tokens[provider].host else provider_domains[provider]
```

Host doesn't always exist so we should default to the provider domains above

malhotra5 commented:
Thanks you for this! I've added one comment but besides that this LGTM

littlebai3618 commented:
> Thanks you for this! I've added one comment but besides that this LGTM

I've updated and resubmitted the changes to /runtime/base.py based on your feedback.

littlebai3618 commented:
fix lint error

```shell
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...........................................(no files to check)Skipped
debug statements (python)................................................Passed
pyproject-fmt........................................(no files to check)Skipped
Validate pyproject.toml..............................(no files to check)Skipped
ruff.....................................................................Passed
ruff-format..............................................................Passed
mypy.....................................................................Passed
[main 65a5f5eb] style: apply ruff formatting to utils.py
```
